### PR TITLE
week3nodejs

### PR DESCRIPTION
**Pull request by alaashhadat**

### CODE REVIEW

File: week3/homework/src/actions.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+function get() {
+  return getToDos().value();
+}
+function getItem(id) {
+  return getTodo(id).value();
+}
+
+function done(id) {
+  return db
+    .get('todos')
```
Comment (eidosam) [inline week3/homework/src/actions.js]: You already created a function to read the todos from the database, so you could replace `db.get('todos')` with `getToDos()`. The same note applies on all functions below.

File: week3/homework/src/getById.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const requireActions = require('./actions');
+function getById(req, res) {
+  if (requireActions.getItem(req.params.id)) {
+    res.status(206);
```
Comment (eidosam) [inline week3/homework/src/getById.js]: Shouldn't be 200? See https://www.restapitutorial.com/httpstatuscodes.html

File: week3/homework/src/getById.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    res.json(requireActions.getItem(req.params.id));
+  }
+  // eslint-disable-next-line indent
+  else {
+    res.status(400);
+    res.json('name a value by id ');
```
Comment (eidosam) [inline week3/homework/src/getById.js]: `res.send`

File: week3/homework/src/markAsDone.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+/* eslint-disable indent */
+'use strict';
+
+const requireFunc = require('./actions');
+
+function markAsDone(req, res) {
+  if (requireFunc.getItem(req.params.id)) {
+    res.status(206);
```
Comment (eidosam) [inline week3/homework/src/markAsDone.js]: Shouldn't be 200?

File: week3/homework/src/markAsNotDone.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+/* eslint-disable indent */
+const requireFunc = require('./actions');
+
+const markAsNotDone = (req, res) => {
+  if (requireFunc.getItem(req.params.id)) {
+    res.status(206);
```
Comment (eidosam) [inline week3/homework/src/markAsNotDone.js]: 200

File: week3/homework/src/markAsNotDone.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    res.json(requireFunc.notDone(req.params.id) && requireFunc.get());
+  }
+ else {
+    res.status(400);
+    res.json('Please define a valid id!');
```
Comment (eidosam) [inline week3/homework/src/markAsNotDone.js]: `res.send`

File: week3/homework/src/removeToDo.js [hunk 7]
```diff
@@ -0,0 +1,18 @@
+const requireActions = require('./actions');
+function removeToDo(req, res) {
+  if (requireActions.getItem(req.params.id)) {
+    res.status(200);
+    res.json(
+      (() => {
+        requireActions.removeToDo(req.params.id);
+        return requireActions.get();
+      })()
+    );
```
Comment (eidosam) [inline week3/homework/src/removeToDo.js]: You could use the same short-circuit you used in the previous functions, for consistency
    
    res.json(requireActions.removeToDo(req.params.id) && requireActions.get());

File: week3/homework/src/actions.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const low = require('lowdb');
```
Comment (eidosam) [inline week3/homework/src/actions.js]: Nice choice!

**Comment by wouterkleijn** [comment]
Thanks for the feedback @eidosam Great! @alaashhadat Can you address his feedback? Thanks!

File: week3/homework/src/actions.js (hunk 1 above)
Comment (alaashhadat) [reply week3/homework/src/actions.js]: Thank you for feedback and for your advices.

**Comment by wouterkleijn** [comment]
Thanks @alaashhadat

**Pull request closed without merging**